samcli/local/init/templates/cookiecutter-aws-sam-hello-ruby/{{cookiecutter.project_name}}/README.md

```diff
@@ -7,7 +7,8 @@ This is a sample template for {{ cookiecutter.project_name }} - Below is a brief
 ├── README.md                   <-- This instructions file
 ├── hello_world                 <-- Source code for a lambda function
 │   ├── app.rb                  <-- Lambda function code
-├── Gemfile                     <-- Ruby dependencies
+│   ├── Gemfile                 <-- Ruby function dependencies
+├── Gemfile                     <-- Ruby test/documentation dependencies
```

Comment:
Not a huge fan of the Double `Gemfile`. How do I know or remember which one to update? Is this absolutely needed? For Node, we had tests under the function directory and excluded them through `.npmignore`

Reply:
So...I'm open to ways we can support this through `sam build`. If we had to pick one, we would not include the global test/docs Gemfile, but then the way the example code does testing wouldn't work easily. I don't know how to get SAM CLI to use a Gemfile in the root directory to build vendored dependencies into each function directory in a single command.

It's actually very possible that multiple Gemfiles may be a reality, and that we just need to teach the paradigm. Why?

1. Layers will eventually be a part of this build story, and you only want to specify those dependencies within the layer itself. In fact, we may want builds where we say "don't fetch all dependencies I need, expect some to be resolved in a layer" down the road.
2. I may not want all of my dependencies in every function.
3. I may want global dependencies in my build structure.

There is actually precedent for this, the Ruby SDK repo itself: https://github.com/aws/aws-sdk-ruby/

We have a global Gemfile for testing, building, and documentation generation. Within each modular package, we have its own gemspec for only the required dependencies. This may be the correct pattern for modular Lambda functions.

Reply:
I can image an appropriate alternative approach, perhaps where we use "groups" per function to keep dependency versions in sync in a global Gemfile. In an ideal world, we could provide options to the user.